frontend/packages/core/src/stepper.tsx

```diff
@@ -206,18 +218,31 @@ const Stepper = ({ activeStep, orientation = "horizontal", children }: StepperPr
           index: idx + 1,
           variant: "pending" as StepIconVariant,
         };
-        if (idx === activeStep) {
+
+        if (completed) {
+          if (completed[idx]) {
+            stepProps.variant = "success";
+          } else if (idx === activeStep) {
+            stepProps.variant = step.props.error ? "failed" : "active";
+          }
+        } else if (idx === activeStep) {
           stepProps.variant = step.props.error ? "failed" : "active";
         } else if (idx < activeStep) {
           stepProps.variant = "success";
         }
 
+        const label = step.props.label ?? `Step ${idx + 1}`;
+        const icon = <StepIcon {...stepProps} />;
```

Comment:
Not sure if it's this change or not but when running locally it seems the number isn't centered on at least the horizontal view. So we'll need to verify that.

Reply:
Might've been a cache problem since I'm not seeing it anymore. Easy enough :)